diag_manager/fms_diag_yaml.F90

```diff
@@ -42,9 +40,112 @@ module fms_diag_yaml_mod
 
 public :: diag_yaml_object_init, diag_yaml_object_end
 public :: diagYamlObject_type, get_diag_yaml_obj, get_title, get_basedate, get_diag_files, get_diag_fields
+public :: diagYamlFiles_type, diagYamlFilesVar_type
 !> @}
 
 integer, parameter :: basedate_size = 6
+integer, parameter :: NUM_SUB_REGION_ARRAY = 8
+integer, parameter :: MAX_STR_LEN = 255
+
+!> @brief type to hold the sub region information about a file
+type subRegion_type
+  character (len=:), allocatable :: grid_type !< Flag indicating the type of region,
+                                              !! acceptable values are "latlon" and "index"
+  real, allocatable :: lat_lon_sub_region (:) !< Array that stores the grid point bounds for the sub region
+                                              !! to use if grid_type is set to "latlon"
+                                              !! [dim1_begin, dim1_end, dim2_begin, dim2_end,
+                                              !!  dim3_begin, dim3_end, dim4_begin, dim4_end]
+  integer, allocatable :: index_sub_region (:) !< Array that stores the index bounds for the sub region to
+                                               !! to use if grid_type is set to "index"
+                                               !! [dim1_begin, dim1_end, dim2_begin, dim2_end,
+                                               !!  dim3_begin, dim3_end, dim4_begin, dim4_end]
+  integer :: tile !< Tile number of the sub region, required if using the "index" grid type
+
+end type subRegion_type
+
+!> @brief type to hold the diag_file information
+type diagYamlFiles_type
+  character (len=:), private, allocatable :: file_fname !< file name
+  character (len=:), private, allocatable :: file_frequnit !< the frequency unit
+  integer, private    :: file_freq !< the frequency of data
+  character (len=:), private, allocatable :: file_timeunit !< The unit of time
+  character (len=:), private, allocatable :: file_unlimdim !< The name of the unlimited dimension
+  logical, private :: file_write !< false if the user doesn't want to the file to be created
+  character (len=:), private, allocatable :: string_file_write !< false if the user doesn’t want the file to be
```

Comment:
Is a string version of file_write really needed? If so, then every place file_write is set, string_file_write needs also be set.
Might it be safer or easier to just use a funcion like : get_file_write_str() if (file_write .eq. .true) return 'true' else return 'false' ?